src/include/executor/execdesc.h

```diff
@@ -23,6 +23,14 @@
 
 struct CdbExplain_ShowStatCtx;  /* private, in "cdb/cdbexplain.c" */
 
+typedef struct YagpHooksQueryState
```

Comment:
It's not really a state, it's a key. So the minimum change would be `YagpHooksQueryKey`, but I'd even go furhter and call it `QueryDescKey` so that it's more neutral and to the point.